binnacle/lib/screens/map_screen.dart

```diff
@@ -0,0 +1,148 @@
+                onMapCreated: _onMapCreated,
+                onTap: _onMapTapped,
+                initialCameraPosition: CameraPosition(
+                  target: snapshot.data.latlng,
+                  zoom: 15.0,
+                ),
+                markers: _markers,
+                polylines: Set<Polyline>.of(_polylines.values),
+              );
+            }
+          }),
+      floatingActionButton: new FloatingActionButton(
+        onPressed: () {
+          setState(() {
+            _isPlacingPoints = !_isPlacingPoints;
+          });
+
+          if (_isPlacingPoints) {
+            // TODO: This would be nicer if it calls the navigators sliding window
+            Scaffold.of(_context).showSnackBar(new SnackBar(
+              content: new Text("Tap the map for where you want to go"),
+            ));
+          } else {
+            _onCourseCreationFinished();
+          }
+        },
+        child: _isPlacingPoints
+            ? Icon(Icons.assistant_photo, size: 36.0)
+            : Icon(Icons.add_location, size: 36.0),
+        // child: Icon(Icons.assistant_photo)
+      ),
+      floatingActionButtonLocation: FloatingActionButtonLocation.centerFloat,
+    ));
+  }
+
+  void _initStartMarker(LatLng startPosition) {
+    _markers.add(Marker(
+      // This marker id can be anything that uniquely identifies each marker.
+      markerId: MarkerId("Start"),
+      position: startPosition,
+      infoWindow: InfoWindow(title: 'Starting point', snippet: 'You are here'),
+      icon: BitmapDescriptor.defaultMarker,
+    ));
+  }
+
+  void _onMapCreated(GoogleMapController controller) {
+    _controller.complete(controller);
+  }
+
+  void _onMapTapped(LatLng position) {
+    if (!_isPlacingPoints) {
+      return;
+    }
+
+    if (_markers.length > MAX_MARKERS) {
+      // TODO: Use the navigator sliding window
+      // BUG: This doesn't work
+      Scaffold.of(context).showSnackBar(new SnackBar(
+        content: new Text("Max number of markers added"),
+      ));
+
+      print('No more points can be placed');
+    }
+
+    // TODO: Use a custom icon instead of the default Google Map one
+    Marker marker = new Marker(
+        markerId: MarkerId(_markers.length.toString()),
+        position: position,
+        onTap: _onMarkerTap,
+        icon: BitmapDescriptor.defaultMarker);
+
+    setState(() {
+      _markers.add(marker);
+    });
+  }
+
+  void _onMarkerTap() {
+    print('the marker was tapped');
+  }
+
+  void _onCourseCreationFinished() {
+    List<LatLng> points = new List();
+    _markers.forEach((marker) => points.add(marker.position));
+
+    print('creating course');
+    _initCourse(points);
+  }
+
+  /// Constructs the polylines that will be shown on the map as the
+  /// course to be sailed. Pretty much connects the dots
+  /// List<LatLng> [points] are a list of geographical points
+  /// returns void as it sets the current state to have these polylines
+  void _initCourse(List<LatLng> points) {
```

Comment:
Again only one point is placed and the start is wherever the phone is when it starts a route

Reply:
Same: @dan-rds is this okay after showing you the demo last night?

Follow up: Should the start point be updated whenever the bloc updates? eg use the current position as the start point instead of when the map was opened. An issue I see with this is the added complexity of ensure start point stays the same once a user is finished inputting their course.